crates/data_components/src/graphql/client.rs

```diff
@@ -1202,6 +1207,65 @@ impl GraphQLClient {
 
         builder.build()
     }
+
+    /// Executes a GraphQL query with page-level retry for transient errors.
+    ///
+    /// This function wraps `execute()` with retry logic using exponential backoff.
+    /// It will retry up to `PAGE_RETRY_MAX_ATTEMPTS` times for retriable errors
+    /// (e.g., 502, 503, 504, timeouts).
+    async fn execute_with_retry(
```

Comment:
This should use the retry utility we have: https://github.com/spiceai/spiceai/blob/bdac31b58005c67df3252f5a68d24e51f2ddc9fe/crates/util/src/lib.rs#L28 with the fibonacci retry policy